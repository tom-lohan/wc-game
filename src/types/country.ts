export type Country = {
	name: string,
	emoji: string,
	group: string,
	ranking: number,
	id: string,
}