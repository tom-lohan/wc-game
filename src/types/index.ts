export type { Country } from './country';
export type { Player } from './player';
export { PageType } from './pageType';