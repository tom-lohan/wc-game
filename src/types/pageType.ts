export enum PageType {
	LANDING_PAGE = 'landing_page',
	ADD_PLAYERS_PAGE = 'add_players_page',
	MAKE_DRAW_PAGE = 'make_draw',
}