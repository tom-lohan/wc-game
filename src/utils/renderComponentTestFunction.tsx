import { configureStore } from '@reduxjs/toolkit'
import { render } from '@testing-library/react'
import React, { ReactNode } from 'react'
import { Provider } from 'react-redux'
import '@testing-library/jest-dom'
import { WCGInitialState } from '../wcgSlice'
import gameReducer from '../wcgSlice'

export const renderComponentTestFunction = (
    children: ReactNode,
    modifiedStore: object = {}
) => {
    const store = {
        wcgame: { ...WCGInitialState },
    }

    const mergedStore = { ...store, ...modifiedStore }

    const mockedStore = configureStore({
        reducer: {
            wcgame: gameReducer,
        },
        preloadedState: mergedStore,
        middleware: (getDefaultMiddleware) =>
            getDefaultMiddleware({ serializableCheck: false }),
    })

    return render(<Provider store={mockedStore}>{children}</Provider>)
}
