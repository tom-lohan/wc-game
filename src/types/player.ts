import { Country } from "./country"

export type Player = {
	name: string,
	email?: string,
	country?: Country,
}