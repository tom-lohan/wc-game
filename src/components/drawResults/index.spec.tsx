import { DrawResults } from '.'
import { Player } from '../../types'
import { renderComponentTestFunction } from '../../utils/renderComponentTestFunction'

const mockedRedraw = jest.fn()

describe('CountryList', () => {
    let mockedPlayers: Player[] = []
    beforeEach(() => {
        mockedPlayers = [
            {
                name: 'Tom',
                country: {
                    name: 'Fakeland',
                    emoji: 'ðŸ˜Ž',
                    ranking: 1,
                    group: 'A',
                    id: 'abc',
                },
            },
            {
                name: 'Mary',
                country: {
                    name: 'MadeUpLand',
                    emoji: 'ðŸ¥¸',
                    ranking: 2,
                    group: 'A',
                    id: 'def',
                },
            },
            {
                name: 'Joe',
                country: {
                    name: 'Imaginaria',
                    emoji: 'ðŸ¤ ',
                    ranking: 3,
                    group: 'A',
                    id: 'ghi',
                },
            },
        ]
        jest.clearAllMocks()
        jest.restoreAllMocks()
    })

    it('should render correctly', () => {
        const { container } = renderComponentTestFunction(
            <DrawResults players={mockedPlayers} redraw={mockedRedraw} />
        )
        expect(container).toMatchSnapshot()
    })
})
