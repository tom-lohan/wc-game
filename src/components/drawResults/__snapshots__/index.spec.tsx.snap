// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryList should render correctly 1`] = `
<div>
  <h1
    class="text-3xl text-center"
  >
    World Cup 2022
  </h1>
  <div
    class="flex flex-wrap justify-center"
  >
    <div
      class="w-1/3 print:w-1/3"
    >
      <div
        class="bordered border-2 m-2 py-3 text-center"
      >
        <h1
          class="text-lg truncate mx-2"
        >
          Joe
        </h1>
        <p
          class="m-0 text-3xl"
        >
          ðŸ¤ 
        </p>
        <p
          class="m-0"
        >
          Imaginaria
        </p>
      </div>
    </div>
    <div
      class="w-1/3 print:w-1/3"
    >
      <div
        class="bordered border-2 m-2 py-3 text-center"
      >
        <h1
          class="text-lg truncate mx-2"
        >
          Mary
        </h1>
        <p
          class="m-0 text-3xl"
        >
          ðŸ¥¸
        </p>
        <p
          class="m-0"
        >
          MadeUpLand
        </p>
      </div>
    </div>
    <div
      class="w-1/3 print:w-1/3"
    >
      <div
        class="bordered border-2 m-2 py-3 text-center"
      >
        <h1
          class="text-lg truncate mx-2"
        >
          Tom
        </h1>
        <p
          class="m-0 text-3xl"
        >
          ðŸ˜Ž
        </p>
        <p
          class="m-0"
        >
          Fakeland
        </p>
      </div>
    </div>
  </div>
  <div
    class="flex items-center justify-center mt-5 print:hidden"
  >
    <button
      class="ant-btn ant-btn-default ant-btn-lg"
      type="button"
    >
      <span>
        Re-draw
      </span>
    </button>
  </div>
</div>
`;
